cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java

```diff
@@ -310,15 +312,45 @@ public Boolean apply(StreamEvent input) {
    * @param startTime Timestamp in milliseconds to start reading event from (inclusive)
    * @param endTime Timestamp in milliseconds for the last event to read (exclusive)
    * @param limit Maximum number of events to read
+   * @param results Collection for storing the resulting stream events
+   * @param <T> Type of the collection for storing results
+   * @return The same collection object as passed in the {@code results} parameter
+   * @throws IOException If fails to read from stream
+   * @throws StreamNotFoundException If the given stream does not exists
+   */
+  @Deprecated
+  public <T extends Collection<? super StreamEvent>> T getEvents(String streamId, long startTime,
+                                                                 long endTime, int limit, final T results)
+    throws IOException, StreamNotFoundException, UnauthorizedException {
+    getEvents(streamId, startTime, endTime, limit, new Function<StreamEvent, Boolean>() {
+      @Override
+      public Boolean apply(StreamEvent input) {
+        results.add(input);
+        return true;
+      }
+    });
+    return results;
+  }
+
+  /**
+   * Reads events from a stream
+   *
+   * @param streamId ID of the stream
+   * @param start Timestamp in milliseconds or now-xs format to start reading event from (inclusive)
+   * @param end Timestamp in milliseconds or now-xs format for the last event to read (exclusive)
+   * @param limit Maximum number of events to read
    * @param callback Callback to invoke for each stream event read. If the callback function returns {@code false}
    *                 upon invocation, it will stops the reading
    * @throws IOException If fails to read from stream
    * @throws StreamNotFoundException If the given stream does not exists
    */
-  public void getEvents(String streamId, long startTime, long endTime, int limit,
+  public void getEvents(String streamId, String start, String end, int limit,
                         Function<? super StreamEvent, Boolean> callback)
     throws IOException, StreamNotFoundException, UnauthorizedException {
 
+    long startTime = TimeMathParser.parseTime(start, TimeUnit.MILLISECONDS);
```

Comment:
Yes since before this if it was a numeric string it would be in milliseconds but parseTime would return a value in seconds if the string is in "now-xs" format